sync/sync.go

```diff
@@ -239,7 +241,9 @@ func (s *Synchronizer) syncBlocks(syncCtx context.Context) {
 	s.startingBlockNumber = &startingHeight
 
 	pendingSem := make(chan struct{}, 1)
-	go s.pollPending(syncCtx, pendingSem)
+	if !s.readOnlyBlockchain {
```

Comment:
Can we move everything that needs to be initialized/run when the db is read-only to the top of `syncBlocks`, and then insert
```go
if s.readOnlyBlockchain {
        <-streamCtx.Done
        streamCancel()
        return
}
```

Reply:
I wanted to keep using the existing "shutdown path" that sync service followed just to be safe. Pushed a new commit, let me know what you think.

Reply:
I think we can move the `fetchers, verifiers := s.setupWorkers()` call below the `if` statement. Otherwise LGTM.